src/joint_storage.rs

```diff
@@ -9,6 +10,10 @@ use storage;
 
 // use spec::Unit;
 
+lazy_static! {
+    static ref JOINTSTORAGE_MUTEX: Mutex<()> = Mutex::new(());
```

Comment:
the lock should be used outside the `readDependentJointsThatAreReady`, it's should be protect the whole `handle_saved_joints`, just remove the lock, I will take care of it